scalr/data_source_current_run_test.go

```diff
@@ -0,0 +1,115 @@
+package scalr
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	scalr "github.com/scalr/go-scalr"
+)
+
+func TestAccCurrentRun_basic(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                  func() { testAccPreCheck(t) },
+		Providers:                 testAccProviders,
+		PreventPostDestroyRefresh: true,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					os.Unsetenv(currentRunIDEnvVar)
+				},
+				Config: testAccCurrentRunDataSourceConfig(rInt),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckNoResourceAttr("data.scalr_current_run.test", "id"),
+				),
+			},
+
+			{
+				PreConfig: launchRun(fmt.Sprintf("test-env-%d", rInt), fmt.Sprintf("test-ws-%d", rInt)),
+				Config:    testAccCurrentRunDataSourceConfig(rInt),
+				PlanOnly:  true,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet(
+						"data.scalr_current_run.test", "id"),
+					resource.TestCheckResourceAttr(
+						"data.scalr_current_run.test", "workspace_name", fmt.Sprintf("test-ws-%d", rInt)),
+				),
+			},
+		},
+	})
+}
+
+func launchRun(environmentName, workspaceName string) func() {
+	return func() {
+		var environmentID *string
+		scalrClient := testAccProvider.Meta().(*scalr.Client)
+
+		envl, err := scalrClient.Environments.List(ctx)
```

Comment:
I think it is time to implement `scalrClient.Environments.GetbyName` because we do the same in [test_workspaces](https://github.com/Scalr/terraform-provider-scalr/blob/db976f227af48905c96ffeac84e051caccaf3b95/scalr/resource_scalr_workspace_test.go#L305).
@lbn what do you think?

Reply:
I think we should either implement that functionality in the API first or just put a helper wrapper for list+filter in this repo.